packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -1833,10 +1809,7 @@ export default (superClass: Class<Parser>): Class<Parser> =>
     tsParseModuleReference(): N.TsModuleReference {
       return this.tsIsExternalModuleReference()
         ? this.tsParseExternalModuleReference()
-        : this.tsParseEntityName({
-            allowReservedEntityName: false,
-            allowReservedQualifiers: false,
-          });
+        : this.tsParseEntityName(/* allowReservedWords */ false);
```

Comment:
I think this is a TS bug (https://github.com/microsoft/TypeScript/issues/48274), but I decided to match the TS behavior and throw.